internal/core/app_build.go

```diff
@@ -129,12 +148,57 @@ func (op *buildOperation) Upsert(
 }
 
 func (op *buildOperation) Do(ctx context.Context, log hclog.Logger, app *App, _ proto.Message) (interface{}, error) {
+	args := []argmapper.Arg{
+		argmapper.Named("HasRegistry", op.HasRegistry),
+	}
+
+	// If there is a registry defined and it implements RegistryAccess...
+	if op.Registry != nil {
+		if ra, ok := op.Registry.Value.(component.RegistryAccess); ok {
+			fn := ra.AccessInfoFunc()
```

Comment:
Should we not be using `callDynamicFuncAny2` here or whatever that thing is called?